app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDao.java

```diff
@@ -378,7 +377,35 @@ public Survey getSurvey(GuidCreatedOnVersionHolder keys, boolean includeElements
         return new QueryBuilder().setSurvey(keys.getGuid()).setCreatedOn(keys.getCreatedOn())
                 .setSkipElements(!includeElements).getOne();
     }
-    
+
+    /** {@inheritDoc} */
+    @Override
+    public String getSurveyGuidForIdentifier(StudyIdentifier studyId, String surveyId) {
+        // Hash key.
+        DynamoSurvey hashKey = new DynamoSurvey();
+        hashKey.setStudyIdentifier(studyId.getIdentifier());
+
+        // Range key.
+        Condition rangeKeyCondition = new Condition().withComparisonOperator(ComparisonOperator.EQ)
+                .withAttributeValueList(new AttributeValue().withS(surveyId));
+
+        // Construct query.
+        DynamoDBQueryExpression<DynamoSurvey> expression = new DynamoDBQueryExpression<DynamoSurvey>()
+                .withConsistentRead(false)
+                .withHashKeyValues(hashKey)
+                .withRangeKeyCondition("identifier", rangeKeyCondition)
+                .withLimit(1);
```

Comment:
Don't we need to use the Index object if querying using the index you've specified for studyId-identifier?

Reply:
Apparently not, if integration tests are any indication.